lib/tasks/eu_exit_business_finder.rake

```diff
@@ -0,0 +1,108 @@
+  # the old facet values in bulk and write only the unique values that remain.
+  def update_facet_values(subscriber_lists)
+    subscriber_lists.each do |list|
+      links = list.links[:facet_values][:any]
+      updated_links = links.map { |original_value| replace_facet_value(original_value) || original_value }
+      updated_links = (updated_links - EuExitFacetMigrationConfig::facet_values_to_remove).uniq
+      list.write_attribute('links', facet_values: { any: updated_links })
+      list.write_attribute('title', title_from_facets(updated_links))
+      list.save
+    end
+  end
+
+  def replace_facet_value(original_facet_value)
+    EuExitFacetMigrationConfig::facet_values_to_replace
+      .select { |replace_value| replace_value[:old_facet_value] == original_facet_value }
+      .map { |replace_value| replace_value[:new_facet_value] }[0]
+  end
+
+  def title_from_facets(facet_values)
+    subscription_title_prefix = "EU Exit guidance for your business in the following #{'category'.pluralize(facet_values.count)}: "
+    titles = facet_values.map do |facet_value|
+      if EuExitFacetMigrationConfig.facet_value_label_overrides[facet_value].present?
+        "'#{EuExitFacetMigrationConfig.facet_value_label_overrides[facet_value]}'"
+      else
+        content_item_title = GdsApi.publishing_api_v2.get_content(facet_value).to_h['title']
+        "'#{content_item_title}'"
+      end
+    end
+    subscription_title_prefix + titles.join(', ')
+  end
+end
+
+module EuExitFacetMigrationConfig
+  def self.facet_values_to_replace
+    [
+      {
+        "old_facet_value": "9f3476e1-8ff0-455d-a14e-003236b2797c",
+        "new_facet_value": "53f9ce4c-7cbb-447f-bdf1-a9b022896d3a"
+      },
+      {
+        "old_facet_value": "b4e507df-f067-4749-9468-3de120775216",
+        "new_facet_value": "24fd50fa-6619-46ca-96cd-8ce90fa076ce"
+      },
+      {
+        "old_facet_value": "356f46a0-17d3-4ba4-8952-8c02244f9045",
+        "new_facet_value": "18c3892e-8a0e-4884-906e-5938380eceee"
+      },
+      # "34a6edd0-46ea-4a76-ae80-8c96709d4f5" is worth noting as
+      # it's the result of a merge between two facets. The
+      # find-and-replace works the same way regardless.
+      {
+        "old_facet_value": "1e3e8abd-135d-4844-afa8-5c51df3d3c57",
+        "new_facet_value": "34a6edd0-46ea-4a76-ae80-8c96709d4f59"
+      },
+      {
+        "old_facet_value": "c1d8057c-76bf-431c-9ac8-6281a4b7b9ca",
+        "new_facet_value": "34a6edd0-46ea-4a76-ae80-8c96709d4f59"
+      },
+      # "94b3cfe2-af89-4744-b8d7-7fc79edcbc85" is a special case
+      # as it's been split into two facets. We should assume the
+      # user wants to remain subscribed to both.
+      {
+        "old_facet_value": "94b3cfe2-af89-4744-b8d7-7fc79edcbc85",
+        "new_facet_value": "9d54c591-f5ca-4d0c-a484-12d5591987cb"
+      },
+      {
+        "old_facet_value": "94b3cfe2-af89-4744-b8d7-7fc79edcbc85",
+        "new_facet_value": "afd45eef-743a-417d-9245-3eab8322116d"
```

Comment:
I think this might raise an issue, as by the time this hash is processed, "94b3cfe2-af89-4744-b8d7-7fc79edcbc85" will have already been overwritten with 9d54c591-f5ca-4d0c-a484-12d5591987cb".

I think the code needs to be able to handle an array of "new_facet_value", and do the replace at the same time.

Reply:
Whoops - good spot!!

Reply:
Fixed. Added a test.